Add options in function pdf_pagefoot & pdf_pagehead of pdf.lib.php

habot-it opened the pull request:
Hello @eldy,

In pdf_pagehead function :
- Add MAIN_USE_BACKGROUND_ON_PDF_ALPHA option for change opacity of the background
- SetPageMark moved to fixe the same problem of MAIN_USE_BACKGROUND_ON_PDF but for MAIN_ADD_PDF_BACKGROUND too

In pdf_pagefoot function :
- Add PDF_FOOTER_HIDDEN to hide all footer (page number will no hidden)
- Add PDF_FOOTER_PAGE_NUMBER_X & PDF_FOOTER_PAGE_NUMBER_Y to change XY position of the page number

Best regards,
Alban

habot-it committed:
Add options in function pdf_pagefoot & pdf_pagehead of pdf.lib.php

stickler-ci committed:
Fixing style errors.

eldy commented on htdocs/core/lib/pdf.lib.php:
```diff
@@ -724,11 +724,13 @@ function pdf_pagehead(&$pdf, $outputlangs, $page_height)
+			if (getDolGlobalString('MAIN_USE_BACKGROUND_ON_PDF_ALPHA')) { $pdf->SetAlpha($conf->global->MAIN_USE_BACKGROUND_ON_PDF_ALPHA); } // Option for change opacity of background
 			$pdf->Image($filepath, (isset($conf->global->MAIN_USE_BACKGROUND_ON_PDF_X) ? $conf->global->MAIN_USE_BACKGROUND_ON_PDF_X : 0), (isset($conf->global->MAIN_USE_BACKGROUND_ON_PDF_Y) ? $conf->global->MAIN_USE_BACKGROUND_ON_PDF_Y : 0), 0, $page_height);
-			$pdf->SetPageMark(); // This option avoid to have the images missing on some pages
+			if (getDolGlobalString('MAIN_USE_BACKGROUND_ON_PDF_ALPHA')) { $pdf->SetAlpha(1); }
```
There was an important non on the line $pdf->SetPageMark(); 
Do you really need to remove this line ? Doing this make the code when MAIN_USE_BACKGROUND_ON_PDF_ALPHA is off to be different than old one.

eldy commented on htdocs/core/lib/pdf.lib.php:
```diff
@@ -724,11 +724,13 @@ function pdf_pagehead(&$pdf, $outputlangs, $page_height)
 			$pdf->SetAutoPageBreak(1, 0); // Restore pagebreak
 		}
 	}
+	$pdf->SetPageMark(); // This option avoid to have the images missing on some pages
```
Why moving this outside of the if (!empty($conf->global->MAIN_USE_BACKGROUND_ON_PDF)
If this test is wrong nothing is modified, so do we really need to do a SetPageMark ?

habot-it replied:
Hello @eldy,
I deleted my change.

Yes, we need $pdf->SetPageMark(); otherwise the image is not really considered as a background image : the layers are mixed between the background image, the text, the frames, etc

Best regards,
Alban

habot-it replied:
Revision 2 : I confirm that when we use a PDF as a background image, we have the same problem as with MAIN_USE_BACKGROUND_ON_PDF

habot-it committed:
Add options in function pdf_pagefoot & pdf_pagehead of pdf.lib.php - Revision 1

Hello @eldy,

In pdf_pagehead function :

    Add MAIN_USE_BACKGROUND_ON_PDF_ALPHA option for change opacity of the background
     ̶S̶e̶t̶P̶a̶g̶e̶M̶a̶r̶k̶ ̶m̶o̶v̶e̶d̶ ̶t̶o̶ ̶f̶i̶x̶e̶ ̶t̶h̶e̶ ̶s̶a̶m̶e̶ ̶p̶r̶o̶b̶l̶e̶m̶ ̶o̶f̶ ̶M̶A̶I̶N̶_̶U̶S̶E̶_̶B̶A̶C̶K̶G̶R̶O̶U̶N̶D̶_̶O̶N̶_̶P̶D̶F̶ ̶b̶u̶t̶ ̶f̶o̶r̶ ̶M̶A̶I̶N̶_̶A̶D̶D̶_̶P̶D̶F̶_̶B̶A̶C̶K̶G̶R̶O̶U̶N̶D̶ ̶t̶o̶o̶

In pdf_pagefoot function :

    Add PDF_FOOTER_HIDDEN to hide all footer (page number will no hidden)
    Add PDF_FOOTER_PAGE_NUMBER_X & PDF_FOOTER_PAGE_NUMBER_Y to change XY position of the page number

Best regards,
Alban

habot-it committed:
Add options in function pdf_pagefoot & pdf_pagehead of pdf.lib.php - Revision 2

eldy committed:
Update pdf.lib.php

eldy committed:
Update pdf.lib.php

eldy committed:
Update pdf.lib.php

eldy committed:
Merge branch 'develop' into pdf.lib.php

habot-it commented:
Merci @eldy ;)
